### ISSUE TITLE
[question] state of the masterbranch

### ISSUE THREAD

**Issue by mrbase** [cachethq/Cachet#1016]
Hi

Was wondering what the state of the master branch is.

Trying to install it results in errors as the `CreateJobsTable` migration is defined twice. Resolving this issue allows me to finalize the installation.

Next issue is in creating Incidents with Components.

This results in a `The status field is required` error originating from the `component_status` not being set. The problem here is in the js-listener, that listens to the `select changed` event - the selector is incorrect.

When editing an Incident status changes is not saved. Here the problem is in the `IncidentController::editIncidentAction` where the reference to the post field is incorrect `Binput::get('status')` should be `Binput::get('incident.status')`

It might just be me, but shouldn't the master branch be "stable" and development be in separate branches which are merged when stable ?

It's a bit hard to follow the process and to contribute with new features when the offset (master) is unstable :)

Good work on the event/command implementation tho :)

**Comment by jbrooksuk** [comment]
> Was wondering what the state of the master branch is.

It's more stable than it was. We essentially use it as a [development branch](https://github.com/cachethq/Cachet#usage-in-production).

> Trying to install it results in errors as the `CreateJobsTable` migration is defined twice. Resolving this issue allows me to finalize the installation.

Whereabouts is it defined twice? I can only see one.

> This results in a The status field is required error originating from the component_status not being set. The problem here is in the js-listener, that listens to the select changed event - the selector is incorrect.

Please make a PR :)

> It might just be me, but shouldn't the master branch be "stable" and development be in separate branches which are merged when stable ?

Nope, we tag stable releases.

> Good work on the event/command implementation tho :)

Thanks, it was a lot of work :)

**Comment by mrbase** [comment]
The double `CreateJobsTable` is defined in:

* database/migrations/2015_05_24_210939_create_jobs_table.php and
* database/migrations/2015_10_07_092245_create_jobs_table.php

\- at least in the master branch :)

Regarding the js-listener:

I'm not sure if I should change the js code or change the template/form so it's not a nested array (`incident[component_id]` and friends) - which is the desired method ?

If `incident[component_id]` is the way you want to go - shouldn't the add template/form follow this also ?

Missed the part about master being the development branch, sorry.

**Comment by jbrooksuk** [comment]
Huh? We only have one `jobs` table, https://github.com/cachethq/Cachet/tree/master/database/migrations

Well we don't really need the form to be an array anymore.

**Comment by mrbase** [comment]
Hmm ... don't know then - I just pulled the code and ran the migrate command in the master branch - oh well .. :)

Will see if I can find the time to remove the array form the form, and create a pr. on that one.

**Comment by jbrooksuk** [comment]
:+1: in the mean time, I'm closing this :)
